site/content/docs/Advanced/Consistency/_index.md

```diff
@@ -1,69 +1,72 @@
 
-4. If a repository can't be opened but was working recently and maintenance has not run yet, it may be helpful to try to remove (or stash away) the most recently-written index files whose names start with `x` in the reverse timestamp order one by one until the issue is fixed. This will effectively roll back the repository writes to a prior state. Exercise caution when removing the files.
+7. Look at the error that Kopia is throwing. Identify the name(s) of the corrupt blobs. 
 
-5. If the steps above do not help, report your issue on https://kopia.discourse.group or https://slack.kopia.io. Kopia has many low-level data recovery tools, but they should not be used by end users without guidance from developers.
+    * If the name(s) start with `n`, `m`, `l` and `x`, then the corruption is likely with index files. Delete the corrupt blobs, run `kopia index recover --parallel=10 --commit`, and see if that fixes your issue. The [`kopia blob delete` command](../../reference/command-line/advanced/blob-delete/) can be used to delete blobs. If your issue has not been solved, go back and delete the whole directory that shares the name with the blob. For example, if the corrupt blob's name is `xn25_04abe9d83a08a65c09bda238af191587-s91f147a798c55c73110-c1`, then find the directory that starts with `xn25` and delete it. Then, run `kopia index recover --parallel=10 --commit` and see if that fixes your issue. If that still does not fix your issues, try running `kopia index recover --parallel=10 --commit --delete-indexes` and see if that fixes your issue.
 
->NOTE: Since all data corruption cases are unique, it's generally not recommended to attempt fixes recommended to other users even for possibly similar issues, since the particular fix method may not be applicable. 
+    * If the name(s) start with `q`, then the corruption is likely with metadata. Run `kopia snapshot fix invalid-files --parallel=10 --commit` and see if that fixes your issue.
 
+    * If the name(s) start with `p`, then the corruption may be in your indexes, metadata, or in the actual blob that stores your backed up files. First, try `kopia index recover --parallel=10 --commit`. If your issue has not been solved, try `kopia index recover --parallel=10 --commit --delete-indexes`. If you are still having issues, try `kopia snapshot fix invalid-files --parallel=10 --commit`. Finally, if the corrupt is still not fixed, try `kopia snapshot fix invalid-files --parallel=10 --commit --verify-files-percent=100` but note that this will download all your files, so it will take a while.
```

Comment:
Given current v2 index structures, `index recover` is almost never the solution worth trying. We should remove this and probably deprecate the command.

Reply:
In general most of the suggestions (except `kopia snapshot fix invalid-files`) are unsafe when used improperly and they can lead to further corruption, that's why they were not mentioned.

Definitely do not suggest deleting any blobs, since this will only help in very few cases if the root cause is properly understood.

I would say if you have timestamps of 1970 or 2038, the filesystem is likely incompatible with Kopia and deleting will not do much.

Reply:
Also, let's not recommend folks run commands with `--commit` until the amount of damage/deletion is understood. Without this switch, kopia will print what it's going to do without actually rewriting snasphots.

Reply:
I just reverted the repair corruption part to what it was before my proposed edits.